### ISSUE TITLE
Receiver class io.vertx.pgclient.spi.PgDriver does not define or inherit an implementation of the resolved method 'abstract io.vertx.sqlclient.Pool createPool

### ISSUE THREAD

**Issue by guy1699** [hibernate/hibernate-reactive#1102]
Working with postgres 14 and vertx-pg-client 4.2.2 on a spring boot app, I'm getting the following exception when calling to:
` EntityManagerFactory emf = Persistence.createEntityManagerFactory(PERSISTENCE_UNIT_NAME, hibernateReactiveProperties());`

```
Caused by: java.lang.AbstractMethodError: Receiver class io.vertx.pgclient.spi.PgDriver does not define or inherit an implementation of the resolved method 'abstract io.vertx.sqlclient.Pool createPool(io.vertx.core.Vertx, java.util.List, io.vertx.sqlclient.PoolOptions)' of interface io.vertx.sqlclient.spi.Driver.
	at io.vertx.sqlclient.Pool.pool(Pool.java:97) ~[vertx-sql-client-4.2.1.jar:4.2.1]
	at org.hibernate.reactive.pool.impl.DefaultSqlClientPool.createPool(DefaultSqlClientPool.java:183) ~[hibernate-reactive-core-1.1.1.Final.jar:1.1.1.Final]
	at org.hibernate.reactive.pool.impl.DefaultSqlClientPool.createPool(DefaultSqlClientPool.java:165) ~[hibernate-reactive-core-1.1.1.Final.jar:1.1.1.Final]
	at org.hibernate.reactive.pool.impl.DefaultSqlClientPool.start(DefaultSqlClientPool.java:133) ~[hibernate-reactive-core-1.1.1.Final.jar:1.1.1.Final]
	at org.hibernate.service.internal.AbstractServiceRegistryImpl.startService(AbstractServiceRegistryImpl.java:344) ~[hibernate-core-5.6.2.Final.jar:5.6.2.Final]
	at org.hibernate.service.internal.AbstractServiceRegistryImpl.initializeService(AbstractServiceRegistryImpl.java:249) ~[hibernate-core-5.6.2.Final.jar:5.6.2.Final]
	at org.hibernate.service.internal.AbstractServiceRegistryImpl.getService(AbstractServiceRegistryImpl.java:214) ~[hibernate-core-5.6.2.Final.jar:5.6.2.Final]
	at org.hibernate.reactive.provider.service.ReactiveImprovedExtractionContextImpl.<init>(ReactiveImprovedExtractionContextImpl.java:69) ~[hibernate-reactive-core-1.1.1.Final.jar:1.1.1.Final]
	at org.hibernate.reactive.provider.service.ReactiveSchemaManagementTool$ReactiveExtractionTool.createExtractionContext(ReactiveSchemaManagementTool.java:52) ~[hibernate-reactive-core-1.1.1.Final.jar:1.1.1.Final]
	at org.hibernate.tool.schema.extract.internal.DatabaseInformationImpl.<init>(DatabaseInformationImpl.java:49) ~[hibernate-core-5.6.2.Final.jar:5.6.2.Final]
	at org.hibernate.tool.schema.internal.Helper.buildDatabaseInformation(Helper.java:183) ~[hibernate-core-5.6.2.Final.jar:5.6.2.Final]
	at org.hibernate.tool.schema.internal.AbstractSchemaValidator.doValidation(AbstractSchemaValidator.java:68) ~[hibernate-core-5.6.2.Final.jar:5.6.2.Final]
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:200) ~[hibernate-core-5.6.2.Final.jar:5.6.2.Final]
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:81) ~[hibernate-core-5.6.2.Final.jar:5.6.2.Final]
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:335) ~[hibernate-core-5.6.2.Final.jar:5.6.2.Final]
	at org.hibernate.reactive.session.impl.ReactiveSessionFactoryImpl.<init>(ReactiveSessionFactoryImpl.java:33) ~[hibernate-reactive-core-1.1.1.Final.jar:1.1.1.Final]
	at org.hibernate.reactive.provider.service.ReactiveSessionFactoryBuilder.build(ReactiveSessionFactoryBuilder.java:52) ~[hibernate-reactive-core-1.1.1.Final.jar:1.1.1.Final]
	at org.hibernate.reactive.provider.impl.ReactiveEntityManagerFactoryBuilder.build(ReactiveEntityManagerFactoryBuilder.java:63) ~[hibernate-reactive-core-1.1.1.Final.jar:1.1.1.Final]
	at org.hibernate.reactive.provider.ReactivePersistenceProvider.createEntityManagerFactory(ReactivePersistenceProvider.java:53) ~[hibernate-reactive-core-1.1.1.Final.jar:1.1.1.Final]
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:79) ~[javax.persistence-api-2.2.jar:2.2]
```

My pom file's dependencies:

```
 <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-webflux</artifactId>
            <version>2.5.0</version>
        </dependency>

        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-rest</artifactId>
            <version>${org.springframework.version}</version>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
            <version>${org.springframework.version}</version>
        </dependency>
        <dependency>
            <groupId>com.slack.api</groupId>
            <artifactId>bolt</artifactId>
            <version>${slack.sdk.version}</version>
        </dependency>
        <dependency>
            <groupId>com.slack.api</groupId>
            <artifactId>bolt-servlet</artifactId>
            <version>${slack.sdk.version}</version>
        </dependency>
        <dependency>
            <groupId>com.google.guava</groupId>
            <artifactId>guava</artifactId>
            <version>${guava.version}</version>
        </dependency>
        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <version>${junit.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.hibernate.reactive</groupId>
            <artifactId>hibernate-reactive-core</artifactId>
            <version>1.1.1.Final</version>
        </dependency>
        <dependency>
            <groupId>io.vertx</groupId>
            <artifactId>vertx-pg-client</artifactId>
            <version>4.2.2</version>
        </dependency>
        <dependency>
            <groupId>com.ongres.scram</groupId>
            <artifactId>client</artifactId>
            <version>2.1</version>
            <optional>true</optional>
        </dependency>
        <dependency>
            <groupId>org.hibernate</groupId>
            <artifactId>hibernate-core</artifactId>
            <version>5.6.2.Final</version>
        </dependency>
        <dependency>
            <groupId>org.hibernate</groupId>
            <artifactId>hibernate-entitymanager</artifactId>
            <version>5.6.2.Final</version>
        </dependency>
        <dependency>
            <groupId>io.smallrye.reactive</groupId>
            <artifactId>smallrye-mutiny-vertx-core</artifactId>
            <version>${mutiny-vertx.version}</version>
        </dependency>
        <dependency>
            <groupId>io.smallrye.reactive</groupId>
            <artifactId>smallrye-mutiny-vertx-pg-client</artifactId>
            <version>${mutiny-vertx.version}</version>
        </dependency>
        <dependency>
            <groupId>io.smallrye.reactive</groupId>
            <artifactId>mutiny-reactor</artifactId>
            <version>1.2.0</version>
        </dependency>
        <dependency>
            <groupId>io.vertx</groupId>
            <artifactId>vertx-core</artifactId>
            <version>4.2.2</version>
        </dependency>

<!--        <dependency>-->
<!--            <groupId>io.vertx</groupId>-->
<!--            <artifactId>vertx-jdbc-client</artifactId>-->
<!--            <version>4.2.2</version>-->
<!--        </dependency>-->

        <dependency>
            <groupId>org.springframework</groupId>
            <artifactId>spring-orm</artifactId>
            <version>5.3.12</version>
        </dependency>
        <dependency>
            <groupId>org.apache.tomcat</groupId>
            <artifactId>tomcat-dbcp</artifactId>
            <version>9.0.1</version>
        </dependency>
        <dependency>
            <groupId>org.postgresql</groupId>
            <artifactId>postgresql</artifactId>
            <version>42.2.23</version>

            <exclusions>
                <exclusion>
                    <groupId>org.slf4j</groupId>
                    <artifactId>slf4j-simple</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-lang3</artifactId>
            <version>3.12.0</version>
        </dependency>
        <dependency>
            <groupId>org.springframework</groupId>
            <artifactId>spring-test</artifactId>
            <version>5.3.13</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.mockito</groupId>
            <artifactId>mockito-inline</artifactId>
            <version>3.8.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.mockito</groupId>
            <artifactId>mockito-core</artifactId>
            <version>3.9.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>joda-time</groupId>
            <artifactId>joda-time</artifactId>
            <version>2.10.13</version>
        </dependency>
    </dependencies>
```

**Comment by DavideD** [comment]
Does it work with Vert.x `4.2.1`?

We tried to release Hibernate Reactive `1.1.2.Final` last week that should work with Vert.x `4.2.2` but it seems that something went wrong. We will try again shortly.

**Comment by guy1699** [comment]
Vert.x 4.2.1 work !!

**Comment by DavideD** [comment]
This should have been fixed by the `1.1.2.Final` release that uses Vert.x `4.2.3`
